src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import core.basesyntax.exceptions.ArrayListIndexOutOfBoundsException;
+import core.basesyntax.exceptions.NoSuchElementException;
+import core.basesyntax.interfaces.List;
+import java.util.Arrays;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double SIZE_INCREASE_FACTOR = 1.5;
+    private static final int INITIAL_SIZE = 0;
+    private int currentCapacity;
+    private int size;
+    private T[] arrayList;
+
+    public ArrayList() {
+        this.arrayList = (T[]) new Object[INITIAL_CAPACITY];
+        this.size = INITIAL_SIZE;
+        this.currentCapacity = INITIAL_CAPACITY;
+    }
```

Comment:
currentCapacity == arrayList.length, so, it is data duplication
```suggestion
    private int size;
    private T[] arrayList;

    public ArrayList() {
        this.arrayList = (T[]) new Object[INITIAL_CAPACITY];
    }
```

Reply:
Ok